crates/cubecl-common/src/kernel.rs

```diff
@@ -59,3 +61,14 @@ pub enum ExecutionMode {
     /// Unchecked kernels are unsafe.
     Unchecked,
 }
+
+/// Implement this trait to create a [kernel definition](KernelDefinition).
+pub trait Kernel: Send + Sync + 'static {
```

Comment:
It's strange that the kernel can't be compiled. Could this be `KernelMetadata` or something like that.